Documentation for other types in collection `requirements.yml`

lilatomic opened the pull request:
##### SUMMARY
Adds documentation on the `file`, `dir`, and `subdirs` types of requirements in a collection's `requirements.yml`

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
`requirements.yml`
`lib/ansible/galaxy/dependency_resolution/dataclasses.py`

##### ADDITIONAL INFORMATION

lilatomic committed:
add missing collection types

(also make the order match the source)

lilatomic committed:
documentation for ``file``, ``dir``, and ``subdirs`` types

samccann commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -42,6 +42,11 @@ Installing a collection from a git repository
+-----------------------------------------
+
+.. include:: ../shared_snippets/installing_collections_file.rst
+
```
Can you make a similar addition to https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/galaxy/user_guide.rst? We cover this info in two places, which is why we use the shared snippets for both of them.

samccann commented on docs/docsite/rst/shared_snippets/installing_collections_file.rst:
```diff
@@ -0,0 +1,24 @@
+      - source: ./my_namespace/
+        type: subdirs
+
+Ansible can also install a collection collected with `ansible-galaxy collection build` by specifying the output file directly:
```
```suggestion
Ansible can also install a collection collected with ``ansible-galaxy collection build`` by specifying the output file directly.
```

samccann reviewed (commented):
@lilatomic sorry for the delay on this PR. It looks like a great addition. I put in a couple of comments to extend this info further.

lilatomic committed:
link installing a collection from files into galaxy guide

also mentions that the tarball can be obtained from Galaxy too

samccann commented on docs/docsite/rst/user_guide/collections_using.rst:
```diff
@@ -42,6 +42,11 @@ Installing a collection from a git repository
+-----------------------------------------
+
+.. include:: ../shared_snippets/installing_collections_file.rst
+
```
Hey @lilatomic sorry I didn't notice this before. Can you move this new section below the `Install multiple collections with a requirements file` section please? That way we've introduced the requirements file before talking about your new examples.

lilatomic replied:
good call, makes sense. I reordered the (applicable) sections in user_guide/collections_using to follow those of galaxy/user_guide, lmk if that's good.

lilatomic committed:
reorder sections

user_guide/collections_using follows galaxy/user_guide
for all sections that apply

samccann commented:
Thanks @lilatomic for the Ansible docs fix!
